internal/elasticsearch/index/indices/read.go

```diff
@@ -0,0 +1,179 @@
+package indices
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"github.com/elastic/terraform-provider-elasticstack/internal/clients/elasticsearch"
+	"github.com/elastic/terraform-provider-elasticstack/internal/utils"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+// Read refreshes the Terraform state with the latest data.
+func (d *dataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var state dataSourceModel
+
+	// Resolve search attribute
+	var search string
+	diag := req.Config.GetAttribute(ctx, path.Root("search"), &search)
+	resp.Diagnostics.Append(diag...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Call client API
+	indices, sdkDiag := elasticsearch.GetIndices(ctx, &d.client, search)
+	resp.Diagnostics.Append(utils.ConvertSDKDiagnosticsToFramework(sdkDiag)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Map response body to model
+	for indexName, index := range indices {
+		indexState := indexModel{
+			Name:              types.StringValue(indexName),
+			SortField:         types.SetValueMust(types.StringType, []attr.Value{}),
+			SortOrder:         types.ListValueMust(types.StringType, []attr.Value{}),
+			QueryDefaultField: types.SetValueMust(types.StringType, []attr.Value{}),
+		}
+
+		if uuid, ok := index.Settings["index.uuid"].(string); ok {
+			indexState.ID = types.StringValue(uuid)
+		} else {
+			indexState.ID = types.StringValue(indexName)
+		}
+
+		// Map index settings
+		if err := populateIndexFromSettings(ctx, utils.FlattenMap(index.Settings), &indexState); err != nil {
+			resp.Diagnostics.AddError("unable to populate index from settings map", err.Error())
+			return
+		}
+		// TODO: We ideally should set read settings to each field to detect changes
+		// But for now, setting it will cause unexpected diff for the existing clients which use `settings`
```

Comment:
Is this comment still accurate, IIUC `populateIndexFromSettings` will do exactly what this comment is talking about?